vendor/engines/secure_rooms/app/models/secure_rooms/indala_ability.rb

```diff
@@ -0,0 +1,23 @@
+module SecureRooms
+
+  class IndalaAbility
+
+    include CanCan::Ability
+
+    def initialize(user, facility)
+      return unless user && facility
+
+      if user.administrator? || user_has_role_at_facility?(user, facility)
+        can :edit, User
+        can :update, User
+      end
+    end
+
+    private def user_has_role_at_facility?(user, facility)
```

Comment:
I would probably just use `user.operator_of?(facility)` here. There are still a few rogue usages of this in controllers and views, but ideally we restrict usage of `operator_of?` to ability checks.